index.js

```diff
@@ -37,8 +38,13 @@ JSCSFilter.prototype.extensions = ['js'];
 JSCSFilter.prototype.targetExtension = 'js';
 JSCSFilter.prototype.processString = function(content, relativePath) {
   if (this.enabled && !this.bypass) {
-    if (this.rules.excludeFiles && this.rules.excludeFiles.indexOf(relativePath) > -1) {
-      return this.disableTestGenerator ? content : '';
+    if (this.rules.excludeFiles) {
+      var excludeMatch = this.rules.excludeFiles.filter(function(excludeFileMatcher) {
+        return minimatch(relativePath, excludeFileMatcher);
```

Comment:
In general :+1:, but we need to cache the comparisons such that we only run minimatch against the relative path once.  Otherwise we are doing a ton more work when we already know that a given file should be excluded.

For inspiration, I do something similar in broccoli-funnel [here](https://github.com/rwjblue/broccoli-funnel/blob/master/index.js#L186-L195).